content/spin/dynamic-configuration.md

```diff
@@ -135,3 +136,49 @@ Got password test_password
 
 Runtime configuration contains information for the selected config provider, such as the [Vault config provider](#vault-config-provider).
 You can supply runtime configuration by providing a value for the `--runtime-config-file` flag when invoking the `spin up` command.
+
+### Key Value Store Runtime Configuration
+
+Spin provides built-in key-value storage. This storage is backed by an SQLite database embedded in Spin by default. However, the Spin runtime configuration file (runtime-config.toml) can be updated to not only modify the SQLite configuration but also choose to use a different backing store. The only store options are the embedded SQLite database or an external Redis database. The following is an example of how an application's `runtime-config.toml` file can be configured for the single default key-value store option.
+
+```toml
+
+[key_value_store.default]
+type = "spin"
+path = ".spin/sqlite_key_value.db"
+
+```
+
+The following is an example of how an application's `runtime-config.toml` file can be configured to use Redis instead. Note the `type` and `url` values, which are set to `redis` and the URL of the Redis host, respectively:
+
+```rust
+[key_value_store.default]
+type = "redis"
+url = "redis://localhost"
+```
+
+Whilst a single default store may be sufficient for certain application use cases, each Spin application can be configured to support multiple stores of any `type`, as shown in the `runtime-config.toml` file below:
+
+> **Note:** At present, when deploying an application to Fermyon Cloud only the single "default" key-value store is supported. To see more about Spin support on Fermyon Cloud, visit the [limitations documentation](http://localhost:3000/cloud/faq#spin-limitations).
+
+> **Note** 
+
+```toml
+[key_value_store.default]
+type = "spin"
+# This path is optional when you are using default as the store. 
+path = ".spin/sqlite_key_value.db"
+
+# Please note the name has been updated here. This is required to connect to your custom store, 
+# as there is no separate name of ID field we can point to.
+[key_value_store.user_data]
+type = "spin"
+# This path is required because you are using a custom named store. 
```

Comment:
Technically the path is required if you want the store to be persistent but we probably don't want to pile on that extra layer of confusion _grin_